src/ui/uiRegistry.js

```diff
@@ -25,30 +24,33 @@ export function register(kind, Component) {
   class ElementView extends mix(React.Component).with(ImmutableProps) {
 
     static propTypes = {
-      dispatch: React.PropTypes.func.isRequired,
       element: React.PropTypes.object.isRequired
     };
 
+    static contextTypes = {
+      store: React.PropTypes.object
```

Comment:
the store is only visible to the Provider replacemement (could be built into the Engine component). I would still abstract the global dispatching function behind a well defined function type instead of passing in the store here.